### PR TITLE
bpo-32572: Add the ftplib option, overrides the IP address.

### PR DESCRIPTION
**Pull request by studioes**
https://bugs.python.org/issue32572

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/ftplib.py [hunk 1]
```diff
@@ -324,6 +331,9 @@ def makeport(self):
         sock.listen(1)
         port = sock.getsockname()[1] # Get proper port
         host = self.sock.getsockname()[0] # Get proper host
+        if(self.externalip!=''): # Override IP address
```
Comment (matrixise) [inline Lib/ftplib.py]: Could you rewrite your condition with this code?
```python
if self.externalip and socket.inet_aton(self.externalip):
    host = self.externalip
```
Thank you

**Review by matrixise — CHANGES_REQUESTED** [review]

File: Lib/ftplib.py [hunk 2]
```diff
@@ -324,6 +331,8 @@ def makeport(self):
         sock.listen(1)
         port = sock.getsockname()[1] # Get proper port
         host = self.sock.getsockname()[0] # Get proper host
+        if self.externalip and socket.inet_aton(self.externalip):
```
Comment (tiran) [inline Lib/ftplib.py:334]: This breaks IPv6 support. You have to use ``inet_pton`` if available. Or rather default ``externalip`` to ``None`` and check for None.

**Review by tiran — CHANGES_REQUESTED** [review]

File: Lib/ftplib.py (hunk 2 above)
Comment (matrixise) [reply Lib/ftplib.py:334]: good catch, thanks for your review

**Comment by tiran** [comment]
Closing because bpo-32572 was rejected

**Pull request closed without merging**